src/javascript/app_2/App/Components/Routes/route_with_sub_routes.jsx

```diff
@@ -20,15 +21,17 @@ const RouteWithSubRoutes = route => {
             }
             result = <Redirect to={to} />;
         } else {
+            const should_show_login_msg = route.is_authenticated && !Client.isLoggedIn();
+
             result = (
-                (route.is_authenticated && !Client.isLoggedIn()) ? // TODO: update styling of the message below
-                    <div className='login-message-wrapper'>
-                        <div className='message'>
-                            <a href='javascript:;' onClick={redirectToLogin}>{localize('Please login to view this page.')}</a>
-                        </div>
-                    </div>
-                    :
-                    <route.component {...props} routes={route.routes}/>
+                should_show_login_msg && route.keep_component &&
+                    <route.component {...props} routes={route.routes}>
+                        <PleaseLoginMessage onLogin={redirectToLogin} />
+                    </route.component>
+                ||
+                should_show_login_msg && <PleaseLoginMessage onLogin={redirectToLogin} />
```

Comment:
we can pass `route.path`  to determine the icon, but
- portfolio drawer has `/` route


or we can pass `route.component.displayName`, but
- `displayName` of route components start with `inject-`, so we would need to use regex
- we can add file in constants folder that would map names to icon components (not sure we would need it for something else though)
- Ashkan said we might replace it with a single icon for all pages, so it was all for naught